snuba/subscriptions/scheduler_consumer.py

```diff
@@ -248,7 +249,9 @@ def __init__(
 
         self.__mode = mode
 
-        self.__partitions = stream_loader.get_default_topic_spec().partitions_number
+        self.__partitions = get_partition_count(
+            stream_loader.get_default_topic_spec().topic
```

Comment:
Use `get_physical_topic_name(slice_id)`

Reply:
Follow-up: will deploying this break something if I don't also update 

https://github.com/getsentry/snuba/blob/c12f247997b18662288aeecf93616b0e6d3a7cc9/snuba/subscriptions/partitioner.py#L14-L25

which still uses the old topic map?

Reply:
Yeah, I think you need to migrate everything off this topic partition map here too if you want to avoid configuring it

Reply:
Are you sure that this PR can't go out alone? I've started looking into replacing this logic and it requires changing a lot more code. If I can do it safely, I'd rather deploy the current change alone (taking into account slicing + Filippo's comment about a clearer error message)

Reply:
I would prefer for this PR not to go out alone, and now I recall it was intentionally not done in the past.

This PR is essentially saying it is ok for the topic to not be set in `TOPIC_PARTITION_COUNTS` and will no longer error when it is wrong. But it is not really ok and will actually cause more subtle failures unless we replace the logic across the board. Specifically all new subscriptions will be inserted into one partition instead of being properly distributed. We will also break consistency - no more guarantees that all data is written before subscriptions are run if they are being put on different partitions. I'd much rather error in an obvious way even if it is one extra step to fix than risk it being subtly wrong in a way that is hard to spot.